File: backend-go/app/models/token.go
package models

type JWTMeta struct {
	CoreModel
	Issuer     string `json:"issuer"`
	Audience   string `json:"audience"`
	IssuedAt   int64  `json:"issuedAt"`
	Expiration int64  `json:"expiration"`
}

type JWTCreds struct {
	CoreModel
	Subject  string `json:"subject"`
	Username string `json:"username"`
}

type JWTPayload struct {
	JWTMeta
	JWTCreds
}

type AccessToken struct {
	CoreModel
	AccessToken string `json:"accessToken"`
	TokenType   string `json:"tokenType"`
}

File: backend/app/models/token.py
from datetime import datetime, timedelta

from app.core.config import ACCESS_TOKEN_EXPIRE_MINUTES, JWT_AUDIENCE
from app.models.core import CoreModel


class JWTMeta(CoreModel):

    iss: str = "myapp.com"
    aud: str = JWT_AUDIENCE
    iat: float = datetime.timestamp(datetime.utcnow())
    exp: float = datetime.timestamp(datetime.utcnow() + timedelta(minutes=ACCESS_TOKEN_EXPIRE_MINUTES))


class JWTCreds(CoreModel):

    sub: str
    username: str


class JWTPayload(JWTMeta, JWTCreds):

    pass


class AccessToken(CoreModel):

    access_token: str
    token_type: str
